test/f90_correct/src/array_constructor_nested_implied_do.f90

```diff
@@ -43,6 +43,46 @@ PROGRAM stress_test_nested_implied_do
                3, 30, 300, 3, 30, 300, 3, 30, 300,&
                4, 40, 400, 4, 40, 400, 4, 40, 400, 4, 40, 400/
 
+  INTEGER(KIND=1) :: i8, j8   ! TY_BINT
+  INTEGER(KIND=2) :: i16, j16 ! TY_SINT
+  INTEGER(KIND=4) :: i32, j32 ! TY_INT
+  INTEGER(KIND=8) :: i64, j64 ! TY_INT8
+
+  ! Different integers fitting into each other
+  ! Array contents are intentionally oversized to show that a warning rather than an error is thrown
```

Comment:
Consider matching the warnings with check_compilation.py.

Reply:
Sure, I will check what the script is doing and consider how to use it for these checks. Thanks for the hint!

Reply:
BTW, please check the [comment I added in the issue report](https://github.com/flang-compiler/flang/issues/992#issuecomment-780572479). `gfortran` is throwing errors here (or nothing if you use `-fno-range-check`)

Reply:
I added the `check_compilation.py` usage and I manually checked that it passes fine.
But if I "break" the expected messages or put them to a wrong line and manually check that the test failed, I can still see the test suite passing. 
If I "break" the test so it does not compile or run - I can see the failures alright.

For example I "broke" the message printed for line 103, by removing most of its body. I saw the output of the check_compilation.py to be:

```
  0 inform,  36 warnings,   0 severes, 0 fatal for stress_test_nested_implied_do
('Unexpected:', 'warning', '/home/mipas/flang/flang/test/f90_correct/lit/../src/array_constructor_nested_implied_do.f90', '103', 'Integer constant truncated to fit data type: 1')
('Missed:', 'warning', '/home/mipas/flang/flang/test/f90_correct/lit/../src/array_constructor_nested_implied_do.f90', '103', 'Integer const')
FAIL
```

Yet, the suite as a whole did not report this...
Any ideas what I might be doing wrong?

Reply:
Nevermind, must be something with my local setup. CI has just shown me some failures, and when I manually checked the logs I saw them too, even though the suite allegedly passed all tests.

Reply:
@bryanpkc , the tests are passing now with the additional checks added.

Reply:
Thanks.